torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1175,6 +1175,20 @@ const WorkerInfo& TensorPipeAgent::getWorkerInfo(
     GroupMembershipLockGuard guard(groupMembershipMutex_, isStaticGroup_);
     it = workerNameToInfo_.find(workerName);
   }
+  if (!isStaticGroup_ && it == workerNameToInfo_.end()) {
+    std::unique_lock<std::mutex> lock(unknownWorkerMutex_);
+    auto pred = [&]() {
+      bool res = workerNameToInfo_.find(workerName) != workerNameToInfo_.end();
```

Comment:
I will update the PR description once the PR is ready, but this will be used in dynamic RPC where when the worker A calls on worker B which has not joined yet (B is not in `workerNameToInfo_` map), then A will block with a timeout.